src/allmydata/scripts/create_node.py

```diff
@@ -344,6 +350,54 @@ def create_node(config):
         os.mkdir(basedir)
     write_tac(basedir, "client")
 
+    # if we're doing magic-wormhole stuff, do it now
+    if config['join'] is not None:
+        print >>out, "Opening wormhole with code '{}'".format(config['join'])
+
+        wh = wormhole.create(
+            appid=u"tahoe-lafs.org/lafs",
+            relay_url=u"ws://wormhole.leastauthority.com:4000/v1",
+            reactor=reactor,
+        )
+        code = unicode(config['join'])
+        wh.set_code(code)
+        yield wh.get_welcome()
+        print >>out, "Connected to wormhole server"
+
+        intro = {
+            u"abilities": {
+                "client-v1": {},
+            }
+        }
+        wh.send_message(json.dumps(intro))
+
+        server_intro = yield wh.get_message()
+        server_intro = json.loads(server_intro)
+
+        print >>out, "  received server introduction"
+        assert u'abilities' in server_intro
+        assert u'server-v1' in server_intro['abilities']
```

Comment:
Rather than assert, what do you think about doing a regular `if` and presenting a more presentable failure to the user?  The traceback is likely to be meaningless to 99% of potential users.

Also, Tahoe-LAFS should include some documentation explaining what the `server-v1` ability corresponds to.  This will be pretty boring, I think, but it will set the stage for someone to write something interesting for `server-v2`.